builtin/providers/aws/resource_aws_vpc_peering_connection.go

```diff
@@ -90,6 +98,12 @@ func resourceAwsVpcPeeringRead(d *schema.ResourceData, meta interface{}) error {
 
 	pc := pcRaw.(*ec2.VpcPeeringConnection)
 
+	if d.Get("auto_accept").(bool) {
+		resourceVpcPeeringConnectionAccept(ec2conn, pc, d)
+	} else {
+		d.Set("accept_status", pc.Status.Code)
+	}
+
```

Comment:
Can we try re-writing this so that `resourceAwsVpcPeeringRead` is the only place we're calling `d.Set("accept_status")` (at least in this context).

As is, a cursory glance shows setting `accept_status` if `auto_accept` is `false`, simply b/c the other set is in another function. 

Additionally, we ignore the error returned from `resourceVpcPeeringConnectionAccept` here.

I'm thinking something like this:

```go
[...]
// in resourceAwsVpcPeeringRead
code := pc.Status.Code
if v, ok := d.GetOk("auto_accept"); ok {
	updatedCode, err := resourceVpcPeeringConnectionAccept(ec2conn, pc, d.Id()); 
	if err!=nil {
		// error handling
	}
	
  //	set to new Status.code,
	//  or other handling
	code = updatedCode
}

d.Set("accept_status", code)
// remainder of func
[...]

func resourceVpcPeeringConnectionAccept(conn *ec2.EC2, oldPc *ec2.VpcPeeringConnection, id string) (string, error) {
// [...]
// accepts string ID instead of resource 
// returns string code, error
}
```

Or something to that effect..